### PR TITLE
Fix incorrect optional balance in blocks_info RPC

### LINKED ISSUE #623: blocks_info rpc command: balance is not optional anymore
The balance is not optional anymore and it is always returned.
Even if balance=false is set as an option, balance is still returned.
The docs state that balance is optional.

### PR DESCRIPTION
**Pull request by zhyatt**
Resolves #623

### CODE REVIEW

File: docs/commands/rpc-protocol.md [hunk 1]
```diff
@@ -829,10 +829,9 @@ Using the optional `json_block` is recommended since v19.0.
   }
 }
 ```
-**Optional "pending", "source", "balance"**
+**Optional "pending", "source"**
 
 _pending, source: version 9.0+_  
-_balance: version 12.0+_  
 Booleans, false by default. Additionally checks if block is pending, returns source account for receive & open blocks (0 for send & change blocks), and returns the balance of the account at the time of the block.
```
Comment (dsiganos) [inline docs/commands/rpc-protocol.md:835]: There is another mention of balance at the end of this sentence.

File: docs/commands/rpc-protocol.md [hunk 2]
```diff
@@ -829,10 +829,9 @@ Using the optional `json_block` is recommended since v19.0.
   }
 }
 ```
-**Optional "pending", "source", "balance"**
+**Optional "pending", "source"**
 
 _pending, source: version 9.0+_  
-_balance: version 12.0+_
```
Comment (dsiganos) [inline docs/commands/rpc-protocol.md:835]: Technically, we should say that balance was optional between v12 and v??

Comment (dsiganos) [reply docs/commands/rpc-protocol.md:835]: balance stopped being optional at v18.0

File: docs/commands/rpc-protocol.md (hunk 1 above)
Comment (zhyatt) [reply docs/commands/rpc-protocol.md]: Removed that last mention

File: docs/commands/rpc-protocol.md (hunk 2 above)
Comment (zhyatt) [reply docs/commands/rpc-protocol.md:835]: Yeah I think leaving that off makes sense given how long ago V12 was (April 2018 actually, and even V18 is no longer an active release). Also the addition of `balance` in V18 is actually covered above in the summary of the call.

**Review by dsiganos — APPROVED** [review]

**Pull request merged**